lab-edward-smith/test/readFileTest.js

```diff
@@ -0,0 +1,35 @@
+const chai            = require('chai');
+const expect          = chai.expect;
+const chaiAsPromised  = require('chai-as-promised');
+const fs              = require('fs');
+const readFiles       = require('../lib/readFiles');
+const Promise         = require('promise');
+
+chai.use(chaiAsPromised);
+
+describe('files should be read async', function() {
+  var fileData = [];
+  before(function(done) {
+    fs.readFile(__dirname + '/first.txt', function(err, data) {
+      fileData.push(data.toString());
+      fs.readFile(__dirname + '/second.txt', function(err, data) {
+        fileData.push(data.toString());
+        fs.readFile(__dirname + '/third.txt', function(err, data) {
+          fileData.push(data.toString());
+          done();
+        });
+      });
+    });
+  });
+
+  it('should read the files in the order which they were passed', function() {
+    var p1 = readFiles.readFile(__dirname + '/first.txt');
```

Comment:
You should be able to pass an array to your file read module and have it output contents in the same order.